utils/utils.js

```diff
@@ -193,7 +193,14 @@ function getDefaultPageProperties() {
 }
 
 function getReferrer() {
-  return document.referrer || "$direct";
+  // This error handling is in place to avoid accessing dead object(document)
+  const referrer = "$direct";
```

Comment:
A bit more appropriate.
```suggestion
  const defaultReferrer = "$direct";
```